app/assets/javascripts/discourse/app/components/modal/change-owner.js

```diff
@@ -0,0 +1,64 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import DiscourseURL from "discourse/lib/url";
+import I18n from "I18n";
+import Topic from "discourse/models/topic";
+import { isEmpty } from "@ember/utils";
+
+export default class ChangeOwnerModal extends Component {
+  @tracked saving = false;
+  @tracked newOwner = null;
+  topicController = null;
+  topic = null;
+
+  constructor() {
+    super(...arguments);
+    this.topicController = this.args.model.topicController;
```

Comment:
Passing a reference to the whole topicController increases the coupling a lot. It means that any refactoring in the controller could affect this modal's functionality. Instead, it would be better to pass exactly what we need into the model. e.g. from the topic controller it could look something like:

```js
modal.show(..., {
  model: {
    selectedPostsUsername: this.selectedPostsUsername,
    selectedPostIds: this.selectedPostIds,
    selectedPostsCount: this.selectedPostsCount,
    deselectAll: this.deselectAll, // and make sure this function has the `@action` decorator for binding
    ...
})
```